lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -182,7 +207,32 @@ namespace Js
                 if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
                     && (attributes & PropertyEnumerable))
                 {
-                    return currentIndex;
+                    bool propertyShadowed = false;
```

Comment:
But this will slow down the case where you have a plain prototype chain significantly since every enumerable prototype property will have to ask the whole chain if it is shadowed again.   That defeat the purpose of having the bit vector (where now it only have benefit of property that has been shadowed before, and not property that hasn't shadowed.